Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,199 @@
+
+package org.usfirst.frc.team199.robot.subsystems;
+
+import org.usfirst.frc.team199.robot.RobotMap;
+import org.usfirst.frc.team199.robot.commands.TeleopDrive;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.kauailabs.navx.frc.AHRS;
+
+import edu.wpi.first.wpilibj.AnalogGyro;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * An example subsystem.  You can replace me with your own Subsystem.
+ */
+public class Drivetrain extends Subsystem implements PIDOutput, PIDSource{
+	// Put methods for controlling this subsystem
+	// here. Call these from Commands.
+
+	private final Encoder leftEnc = RobotMap.leftEnc;
+	private final Encoder rightEnc = RobotMap.rightEnc;
+	private final WPI_TalonSRX dtLeftDrive = RobotMap.dtLeftDrive;
+	private final WPI_TalonSRX dtRightDrive = RobotMap.dtRightDrive;
+	private final SpeedControllerGroup dtLeft = RobotMap.dtLeft;
+	private final SpeedControllerGroup dtRight = RobotMap.dtRight;
+	private final DifferentialDrive robotDrive = RobotMap.robotDrive;
+	private final PIDController turnController = RobotMap.turnController;
+	private final PIDController moveController = RobotMap.moveController;
+//	private final PIDController moveLeftController = RobotMap.moveLeftController;
+//	private final PIDController moveRightController = RobotMap.moveRightController;
+	
+	private final AHRS ahrs = RobotMap.ahrs;
+	private final AnalogGyro dtGyro = RobotMap.dtGyro;
+	private final DoubleSolenoid dtGear = RobotMap.dtGear;
+	
+	public void pushGear(boolean forw) {
+		if(forw) {
+			dtGear.set(DoubleSolenoid.Value.kForward);
+		} else {
+			dtGear.set(DoubleSolenoid.Value.kReverse);
+		}
+	}
+	public void stopGear() {
+		dtGear.set(DoubleSolenoid.Value.kOff);
+	}
+	
+	private double pidOut = 0;
+	public void resetAHRS() {
+		ahrs.reset();
+	}
+	
+	public void setLeftMotor(double value) {
+		dtLeft.set(value);
+	}
+	private void stopLeftMotor() {
+		dtLeft.stopMotor();
+	}
+	
+	public void setRightMotor(double value) {
+		dtRight.set(value);
+	}
+	private void stopRightMotor() {
+		dtRight.stopMotor();
+	}
+	
+	public void stopDrive() {
+		stopRightMotor();
+		stopLeftMotor();
+	}
+	
+	public void arcadeDrive(double speed, double turn) {
+		robotDrive.arcadeDrive(speed, turn, SmartDashboard.getBoolean("Square Drive Values", false));
+	}
+	public void tankDrive(double leftSpeed, double rightSpeed) {
+		robotDrive.tankDrive(leftSpeed, rightSpeed, SmartDashboard.getBoolean("Square Drive Values", false));
+	}
+	
```

Comment:
can you change the names of all these methods to reflect what they're returning? bc rn it makes it seems like they're returning the actual motor or gyro when they're not

Reply:
Returning the actual object would render making them private pointless, functions should be commented by the author of the tests.

Reply:
@doawelul you seem to have misunderstood @lhmcgann's request. She's asking for clearer method names, not asking for the actual motor/gyro to be returned.

And, per my earlier comment, the person who writes the code is generally responsible for commenting it appropriately.